### PR TITLE
Update loss G_loss_S

### PR DESCRIPTION
**Pull request by jurahel**
In the supervised stage of generator training we train both the Generator and the Supervisor. G_loss_S loss only includes terms involving the supervisor, thus we must exchange the loss function from

            G_loss_S = tf.keras.losses.MSE(H[:, 1:, :], H_hat_supervise[:, 1:, :])
to 

            G_loss_S = tf.keras.losses.MSE(H_hat[:, 1:, :], H_hat_supervise[:, 1:, :])

with 

            H = encoder(X)
            # Generator
            E_hat = generator(X)
            H_hat = supervisor(E_hat)
            H_hat_supervise = supervisor(H)

since the generator is not involved in generating H but in H_hat. Please correct me if I am wrong.

### CODE REVIEW

**Comment by jsyoon0823** [comment]
Actually, it is not true.
Supervised loss is used for training the supervisor.
The supervisor is trained to generate next sequence using the current sequence.
Therefore, the current loss:

G_loss_S = tf.losses.mean_squared_error(H[:,1:,:], H_hat_supervise[:,:-1,:])

is the correct loss function for G_loss_S.
